RESTWS-741: Remove 1.8 submodule (Removed 0mod-1.8Controller)

josephbate opened the pull request:
<!--- Add a pull request title above in this format -->
<!--- real example: 'RESTWS-738 Remove concept property setters from ObsResource classes' -->
<!--- 'RESTWS-JiraIssueNumber JiraIssueTitle' -->
## Description of what I changed
<!--- Describe your changes in detail -->
<!--- It can simply be your commit message, which you must have -->

1. Transfered controller classes and dependant classes from 1.8 to 1.9
2. Edited `RestTestConstants1_9.java`  to include all the constants of `RestTestConstants1_8.java`
3. edited all classes in the omods above 1.8 to import `RestTestConstants1_9.java` 
4. transfered resourse datasets from 1.8 to 1.9

Cc. @dkayiwa @ibacher @mks-d 

## Issue I worked on
<!--- This project only accepts pull requests related to open issues -->
<!--- Want a new feature or change? Discuss it in an issue first -->
<!--- Found a bug? Point us to the issue/or create one so we can reproduce it -->
<!--- Just add the issue number at the end: -->
see https://openmrs.atlassian.net/browse/RESTWS-741

## Checklist: I completed these to help reviewers :)
<!--- Put an `x` in the box if you did the task -->
<!--- If you forgot a task please follow the instructions below -->
- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

ibacher commented on omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/ConceptMapController1_8Test.java:
```diff
@@ -1,108 +0,0 @@
-/**
- * Tests CRUD operations for {@link ConceptMapType}s via web service calls
- */
-public class ConceptMapController1_8Test extends MainResourceControllerTest {
```
Shouldn't this have been migrated?

josephbate replied:
noted 
will correct that

josephbate replied:
Hello @ibacher 
the `ConceptMapController1_8Test` is it the same as testing the `ConceptMapController1_9Test` which the same methods thats is why didn't include it 
the solution i had was to merge them to one test case but i realized they are testing the same methods 
what's your take on this

ibacher replied:
That's fine then. As long as the tests are covered.

ibacher commented:
Also, it doesn't look like this removes the whole submodule?

josephbate commented:
> Also, it doesn't look like this removes the whole submodule?

true i am doing it in phases 
it helps me keep track and easily spot failures.
 next is resources

ibacher commented:
Ah ok! Do you mind if I convert this to a draft then? (I like the idea of doing this in stages!)

josephbate commented:
> Ah ok! Do you mind if I convert this to a draft then? (I like the idea of doing this in stages!)

It's fine with
Should I continue to push more commits  on this same pr or should i make separate ones for each commit?

ibacher commented:
Let's keep it all in one PR, if that's alright...

josephbate commented:
> Let's keep it all in one PR, if that's alright...

Alright then 
Thank you @ibacher

josephbate commented:
hello @ibacher 
here is an update on this ticket , i have managed to migrate the resource classes to from omod 1.8 to 1.9 
however i ran into trouble with some test classes which cause test failures;
following up on the talk thread [Testcase failures](https://talk.openmrs.org/t/test-case-failures-after-transfering-personattributeresource1-8-to-omod-1-9/43488)

1. on  `PersonAttributeResource1_9Test()` should we ignore the `setValue_shouldSetProperAttributableIdIfFound()` or just modify the expected result to match the actual result
2. on `PersonAttributeTypeResourceWithConcept1_9Test()` please in `validateFullRepresentation()` inspite of the correct approach to cast an instance of SimpleObject to an org.openmrs.Concept it still throws a `ClassCastException`
3. i have commented out both of this test cases as i wait for your thouhts on these two issues

Cc. @dkayiwa  @mks-d

ibacher commented:
I've fixed the tests so they work (according to me).

josephbate commented:
> I've fixed the tests so they work (according to me).

thank you for is 
it also makes sense to for 

> I've fixed the tests so they work (according to me).

it makes sense with me too 

let me these changes update my repository and send my last commit
iam gonna make a new pr if you don't mind @ibacher

josephbate commented:
hello @ibacher 
i have created a new pr 

see https://github.com/openmrs/openmrs-module-webservices.rest/pull/624

please look into it i think its ready for mergeing
